plugins/org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/builderState/MarkerUpdaterImpl.java

```diff
@@ -65,12 +65,17 @@ private void processDelta(Delta delta, /* @Nullable */ ResourceSet resourceSet,
 		IMarkerContributor markerContributor = getMarkerContributor(uri);
 		CheckMode normalAndFastMode = CheckMode.NORMAL_AND_FAST;
 
-		for (Pair<IStorage, IProject> pair : mapper.getStorages(uri)) {
+		for (Pair<IStorage, IProject> pair : mapper.getStorages(uri, false)) {
 			if (monitor.isCanceled()) {
 				throw new OperationCanceledException();
 			}
 			if (pair.getFirst() instanceof IFile) {
 				IFile file = (IFile) pair.getFirst();
+				
```

Comment:
For completeness: `file.isReadonly` is by no means an indicator whether a file should be processed or not, e.g. pessimistic VCS will checkout files as readonly by default.